### PR TITLE
Bulk Selector "Select-All" Function Improvements

### PR DESCRIPTION
**Pull request by aprilma419**
### **Description**
Problem(issue #731): There’s no indication for users to understand if the “select-all” selection is only for the current page items or all items across all pages.

Fixed the issue by adding an indication row down below the toolbar after the user clicked the bulk selector, which is to let the user know what the current selection state is.

### CODE REVIEW

File: pattern-library/forms-and-controls/bulk-selector/design/design.md [hunk 1]
```diff
@@ -29,4 +29,32 @@ With the partial selection component, the bulk selector checkbox fill can reflec
...
+ 2. To select all items cross all pages
+ 3. The number of items cross all pages that are selected
+ 4. Deselect all to clear the selection cross pages
+
+
+ ![Bulk Selector](img/BulkSelector-SelectAll-State.png)
+
+
+## Mixed States
+When the user has selected all items on current page or cross all pages, but unchecked one or more items at the same time, it will show the "checked items" of "all selected items" in the indication.
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/bulk-selector/design/design.md]: Change to "When the user has selected all items on the current page or across all pages and then unchecks one or more items, the message will be updated as shown.

File: pattern-library/forms-and-controls/bulk-selector/design/design.md [hunk 2]
```diff
@@ -29,4 +29,32 @@ With the partial selection component, the bulk selector checkbox fill can reflec
...
+
+
+1. Checked item number of total item number on this page
+2. Checked item number of total item number cross all pages
+
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/bulk-selector/design/design.md]: I think you should also note here that the checkbox changes to the intermediate state.

**Comment by aprilma419** [comment]
Hi Matt, I think your comments make sense here, so I re-did and updated the file as you mentioned.  Please check it out and let know if you have other questions. Thanks! @mcarrano

**Comment by lizsurette** [comment]
@aprilma419 @mcarrano - This looks great! I noticed that we are using the term "Deselect All" in a few places, but under the bulk selector dropdown it's "Select None". I remember having a conversation about it previously and am totally fine with this direction that's proposed, but am also wondering if this option should simply read "Deselect All" to be consistent.

**Comment by aprilma419** [comment]
@lizsurette Thanks and I agreed. I think this could work in terms of design consistency.

**Comment by amysueg** [comment]
LGTM!

**Comment by mcarrano** [comment]
Regarding the issue that @lizsurette raised. I am looking at what Google does and they do not use the word "Select" at all in the menu.

screen shot 2018-12-12 at 2 10 03 pm

@aprilma419 what do you think? Should we change "Select none" to Deselect all" in the menu or leave as is. I don't feel strongly. "Deselect all" is probably more correct but not sure which one will have faster user recognition time. Another consideration is that Select None in the menu means only to clear selection on the current page, right? So maybe they should be different. Thoughts anyone?

**Comment by lizsurette** [comment]
> Thoughts anyone?

I would be happy with this direction too! To me the important thing is that the tooltip and the dropdown text are consistent.

**Comment by LHinson** [comment]
looping in @stacymcauliffe for her 2 cents!

**Comment by aprilma419** [comment]
@mcarrano I also noticed that Gmail uses "clear selection" to deselect as you can see below. 

screen shot 2018-12-13 at 11 14 52 am

Right, I forgot that the "select none" in dropdown menu is only for the current page. I think "Deselect all" term works here in the indication situation as it's a bit different from the dropdown menu.

**Comment by mcarrano** [comment]
I am going to proceed to merge this. @aprilma419 thanks for successfully completing your first PattenFly contribution!

**Comment by aprilma419** [comment]
Got it. Thank you Matt @mcarrano and everyone!

**Pull request merged**